test/e2e/common/node/sysctl.go

```diff
@@ -213,15 +192,27 @@ var _ = SIGDescribe("Sysctls [LinuxOnly] [NodeConformance]", func() {
 		ginkgo.By("Creating a pod with an ignorelisted, but not allowlisted sysctl on the node")
 		pod = podClient.Create(pod)
 
-		ginkgo.By("Watching for error events or started pod")
-		// watch for events instead of termination of pod because the kubelet deletes
-		// failed pods without running containers. This would create a race as the pod
-		// might have already been deleted here.
-		ev, err := f.PodClient().WaitForErrorEventOrSuccess(pod)
+		ginkgo.By("Wait for pod failed reason")
+		// watch for pod failed reason instead of termination of pod
+		err := waitForPodFailedReason(f, pod, "SysctlForbidden")
 		framework.ExpectNoError(err)
-
-		ginkgo.By("Checking that the pod was rejected")
-		gomega.Expect(ev).ToNot(gomega.BeNil())
-		framework.ExpectEqual(ev.Reason, "SysctlForbidden")
 	})
 })
+
+// waitForPodFailedReason wait for pod failed reason in status, for example "SysctlForbidden".
+func waitForPodFailedReason(f *framework.Framework, pod *v1.Pod, reason string) error {
```

Comment:
nit: I think we should have this sort of helper func somewhere for DRY

Reply:
Done. Move to test/e2e/framework/pod/wait.go and solve the import cycle.